From 53138a0e32bc00d8d8e792e2df6169879d8a7b08 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 13 Jul 2022 08:19:39 +0300
Subject: [PATCH] [SW-95553] gaudi3: add function for disabling pcie error resp

This patch adds separate function for disabling pcie error resp
for better readability as part of pcie link up routine.

Change-Id: I6d8121957d6ae22ddf5c9045fb0ff02bb15c6f1a
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 14815fe..83656ff 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -110,6 +110,13 @@
 		regmap_writel(0x0, D0_PSOC_BOOT_CONF_D0_PIF_ISO);
 }
 
+static void disable_pcie_error_resp(void)
+{
+	/* Disable PCIe error responses (both HBW and LBW) */
+	regmap_writel(0x1, mmPCIE_RSP_OK);
+	regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
+}
+
 static void rom_pcie_recovery(void)
 {
 	uint32_t pcie_ctrl_val = arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_CONTROL_MASK;
@@ -121,9 +128,7 @@
 	if ((arcboot_rom_global.die_id && pcie_ctrl_val) ||
 		(!arcboot_rom_global.die_id && !pcie_ctrl_val)) {
 		disable_pif_isolation();
-		/* Disable PCIe error responses (both HBW and LBW) */
-		regmap_writel(0x1, mmPCIE_RSP_OK);
-		regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
+		disable_pcie_error_resp();
 		pcie_config();
 	}
 }
